tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -83,26 +83,30 @@ export const Manufacturer = (_, context) => {
   }
 
   // Get a ManufacturableData from a QueueBlueprint using its type, category, and name.
-  let queueBlueprintRefs = data.queue.map((queued:QueueBlueprint) =>
+  let queueBlueprintRefs = data.queue && data.queue.length
+  && data.queue.map((queued:QueueBlueprint) =>
     blueprints_by_category[queued.category].find((key) => (key.name === queued.name))
   );
 
   return (
     <Window width={1200} height={600} title={data.fabricator_name}>
-      {!data.indicators.hasPower && <PowerAlertModal width={100-SETTINGS_WINDOW_WIDTH} height={"100%"} />}
+      {data.indicators && !data.indicators.hasPower
+      && <PowerAlertModal width={100-SETTINGS_WINDOW_WIDTH} height={"100%"} />}
       <Window.Content scrollable>
         <Stack>
           <Stack.Item grow>
             <Section>
-              {!data.indicators.hasPower && <Dimmer />}
-              {data.all_categories.map((category:string) => (
+              {data.indicators && !data.indicators.hasPower && <Dimmer />}
+              {data.all_categories && data.all_categories.length
+              && data.all_categories.map((category:string) => (
                 blueprints_by_category[category].length > 0 && (
                   <Collapsible
                     key={category}
                     open
                     title={`${category} (${blueprints_by_category[category].length})`}
                   >
-                    {blueprints_by_category[category].map((blueprint:ManufacturableData, index:number) => (
+                    {blueprints_by_category[category] && blueprints_by_category[category].length
+                    && blueprints_by_category[category].map((blueprint:ManufacturableData, index:number) => (
```

Comment:
```suggestion
                    {(blueprints_by_category[category] ?? []).map((blueprint, index) => (
```
Should be able to infer the function argument types? If not, `blueprints_by_category` needs better typing.